### DISCUSSION TITLE
Remove background-color: transparent from default button styles

### DISCUSSION THREAD

**Discussion by htutwaiphyoe** [tailwindlabs/tailwindcss#12063] [Ideas]
This background-color: transparent is overwriting other backgroud color CSS of button.

button,
[type='button'],
[type='reset'],
[type='submit'] {
  -webkit-appearance: button; /* 1 */
  background-color: transparent; /* 2 */
  background-image: none; /* 2 */
}

I have to be disabled preflight in config and add default reset CSS in global.css file manually to get my button background color work.
https://unpkg.com/tailwindcss@3.0.24/src/css/preflight.css

**Comment by huykon** [comment]
I have same concern

**Comment by alex-krasikau** [comment]
Do you set other backgrounds using the Tailwind classes, like `bg-gray-800`, or using your custom classes?

If the second is the case, make sure, that you import the Tailwind CSS file first, before your custom CSS file.

**Reply by huykon** [reply]
Hi @alex-krasikau, let me explain my case. I'm writing a react app using tailwind to attach to the a website (NO using tailwind), the website is working normally before my app was attached. so I don't want tailwind override my css in same page. Do you know any way to do it?

**Reply by alex-krasikau** [reply]
In your case disabling preflight may help - https://tailwindcss.com/docs/preflight#disabling-preflight

**Reply by huykon** [reply]
Thanks your way, it's good but I need to code more =))

**Comment by blamarvt** [comment]
For me the problem was really a problem of CSS (?) precedence. On a base install of Tailwind if you have:
```
@tailwind base;
@tailwind components;
@tailwind utilities;

button {
   @apply bg-blue-500;
}
```

You're going to be very frustrated because the background of your `type=submit` buttons are going to remain transparent.

If you change the button style to:
```
button[type="submit"] {
   @apply bg-blue-500
}
```

...then all is well.

It makes sense in retrospect that the most specific definition wins it's just a bit counterintuitive for me!

**Comment by venudyavan** [comment]
This background-color: transparent is overwriting other backgroud color CSS of button.

button,
[type='button'],
[type='reset'],
[type='submit'] {
-webkit-appearance: button; /* 1 /
background-color: transparent; / 2 /
background-image: none; / 2 */
}
corePlugins: {
    preflight: false,
  }
  It is not working

**Comment by Alvindelito** [comment]
I had this issue as well until I moved my custom classes to inside of the `extend` object in `tailwind.config.js`

```javascript
export default {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {
      colors: {
        ivory: "#FFFFEF",
        beige: "#F5F4DB",
        white: "#FFF",
        yellow: "#F7F59D",
        gray: "#D6D5D3",
      },
      fontFamily: {
        'header': ["Chonburi", "serif"],
        'body': ["Montserrat", "sans-serif"]
      }
    },
  },
  plugins: [],
}
```

I am using react vite and did not have to set `preflight` to false

**Reply by bschm0622** [reply]
Anyone know how to fix this in Tailwind 4? facing the same issue....

**Reply by wongjn** [reply]
Consider providing a git repo that reproduces the undesired behavior.

**Reply by Brent-Knigge** [reply]
Here is a button that does not display properly.  I took it directly from the first example here (for React).   https://tailwindcss.com/plus/ui-blocks/application-ui/overlays/modal-dialogs

```
<button
                  type="button"
                  onClick={() => setOpen(false)}
                  className="bg-red-600 inline-flex w-full justify-center rounded-md  px-3 py-2 text-sm font-semibold text-white shadow-xs hover:bg-red-500 sm:ml-3 sm:w-auto"
                >
                  Deactivate
</button>
```

A work-around for this issue is to use the **!** symbol in front of bg-*.  I.e. **!bg-red-600**

Hope this helps

**Reply by rozsazoltan** [reply]
> Anyone know how to fix this in Tailwind 4? facing the same issue....

I don't quite understand what your problem is right now without reproduction. If the basic question is about the discussion, then the answer is using `@layer`:

* https://github.com/tailwindlabs/tailwindcss/discussions/12063#discussioncomment-14944480
* [From v4 the reset style cannot be overridden by TailwindCSS classes](https://stackoverflow.com/q/79441698/15167500)

If the question is about color declaration, then the answer is a CSS-first configuration.
* [How to use custom color themes in TailwindCSS v4](https://stackoverflow.com/a/79499827/15167500)
* [New configuration option in v4](https://stackoverflow.com/a/79383770/15167500)
* [Which TailwindCSS v4 namespace matches a given TailwindCSS v3's theme keys?](https://github.com/tailwindlabs/tailwindcss/discussions/19020)

**Reply by Brent-Knigge** [reply]
If look at the first example here https://tailwindcss.com/plus/ui-blocks/application-ui/overlays/modal-dialogs we have a modal with a couple of buttons.

<img width="594" height="259" alt="image" src="https://github.com/user-attachments/assets/ac4658c0-3799-4dea-8da9-57850fad2522" />

If we take that code and use it, the modal looks like this. 

<img width="561" height="250" alt="image" src="https://github.com/user-attachments/assets/9e4c5a11-8b81-4ab1-a055-062051c4446d" />

This all comes back to people having issues with setting the background color of buttons.  I'm using Tailwind V4, and this is the result I get.

The work-around I found was using **!**.  @rozsazoltan has now enlightened me of the **@layer** and further information about the base layer and the defaults that it uses.  Thank you.

**Comment by rozsazoltan** [comment]
* [From v4 the reset style cannot be overridden by TailwindCSS classes](https://stackoverflow.com/q/79441698/15167500)
```css
@layer base {
  button, [type='button'], [type='reset'], [type='submit'] {
    -webkit-appearance: button;
    background-color: transparent;
    background-image: none;
  }
}
```

* [Why doesn't the last added class override the previous property?](https://stackoverflow.com/a/79725940/15167500)
  * [Overwrite Tailwind classes](https://discord.com/channels/486935104384532500/1390678392910909510/1390683307855052831)
  * [Why doesn't dynamic Tailwind class override static class in my element?](https://stackoverflow.com/questions/79604241/why-doesnt-dynamic-tailwind-class-override-static-class-in-my-element/79605308#79605308)
